app/src/main/java/app/takumi/obayashi/charmme/MainActivity.kt

```diff
@@ -43,34 +45,62 @@ class MainActivity : AppCompatActivity() {
             }
         }
 
+        makeCharmList()
+        setUpGestureLibrary()
+        setUpPaint()
+        timerViewLayout.visibility = INVISIBLE
+
         readGesture.addOnGesturePerformedListener { _, gesture ->
             val predictions: ArrayList<Prediction>? = gestureLibrary?.recognize(gesture)
             val mostLikelyPrediction: Prediction? = predictions?.maxBy { it.score }
-
             val charm = charmList?.find { it.name == mostLikelyPrediction?.name }
-
             startCharmTimer(charm!!)
+
+            if (gesture.strokesCount > 0) {
+                for (stroke in gesture.strokes) {
+                    val path: Path = stroke.path
+                    mCanvas!!.drawPath(path, mPaint!!)
+                }
+                gestureImage.setImageBitmap(mBitmap)
+            }
         }
+
+        readGesture.viewTreeObserver
+            .addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
+                override fun onGlobalLayout() {
+                    readGesture.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                    mBitmap = Bitmap.createBitmap(
+                        readGesture.width,
+                        readGesture.height,
+                        Bitmap.Config.ARGB_8888
+                    )
+                    mCanvas = Canvas(mBitmap!!)
+                }
+            })
```

Comment:
https://developer.android.com/reference/android/view/ViewTreeObserver
http://tech.admax.ninja/2014/09/17/how-to-get-the-height-and-width-of-the-view/